src/test/java/com/puppycrawl/tools/checkstyle/DetailAstImplTest.java

```diff
@@ -510,7 +510,10 @@ else if (file.getName().endsWith(".java")
                     && !file.getName().endsWith("InputGrammar.java")
                     // comment only files, no root
                     && !file.getName().endsWith("InputPackageDeclarationWithCommentOnly.java")
-                    && !file.getName().endsWith("InputSingleSpaceSeparatorEmpty.java")) {
+                    && !file.getName().endsWith("InputSingleSpaceSeparatorEmpty.java")
+                    && !file.getName().endsWith("InputNoCodeInFile1.java")
+                    && !file.getName().endsWith("InputNoCodeInFile2.java")
+                    && !file.getName().endsWith("InputNoCodeInFile3.java")) {
```

Comment:
it is better to extract these endings into some list and iterate over it, it will be more readable

Reply:
Done.